go/vt/wrangler/reparent.go

```diff
@@ -490,11 +487,19 @@ func (wr *Wrangler) plannedReparentShardLocked(ctx context.Context, ev *events.R
 		if alias == masterElectTabletAliasStr {
 			continue
 		}
+		if alias == topoproto.TabletAliasString(oldMasterTabletInfo.Alias) {
```

Comment:
We should make SetMaster idempotent so we can both ensure it synchronously here, with a wait on the reparent journal, as well as keep the background fix-up triggered via the shard record.

Also, I had intentionally removed as much info about who the "old master" (si.MasterAlias) was as possible, because we need to be able to fix the state without knowing that. If PRS is run again after the shard record is updated, we won't know the previous value of si.MasterAlias, so we need to determine all fix-up actions based on the actual state of each tablet, rather than relying on the shard record as we used to.

Reply:
:+1: